persistence/src/vespa/persistence/spi/persistenceprovider.h

```diff
@@ -374,6 +376,16 @@ struct PersistenceProvider
      * source1 and source2 should be stored in the target bucket.
      */
     virtual Result join(const Bucket& source1, const Bucket& source2, const Bucket& target, Context&) = 0;
+
+    /*
+     * Register a listener for updates to resource usage.
+     */
+    virtual void register_resource_usage_listener(std::shared_ptr<IResourceUsageListener> listener) = 0;
```

Comment:
Why do we need ways to register and unregister ?
Couldn't this just be an IResourceUsageListener & in the initialize method that is valid for the lifetime of the provider ?

Reply:
I'll remove the register / unregister methods and add argument to initialize as proposed.